src/sdam/state/mod.rs

```diff
@@ -225,54 +240,55 @@ impl Topology {
         self.common.message_manager.notify_topology_changed();
     }
 
-    /// Updates the topology based on an error that occurs before the handshake has completed during
-    /// an operation.
-    pub(crate) async fn handle_pre_handshake_error(&self, error: String, server: &Server) -> bool {
+    pub(crate) async fn handle_application_error(
```

Comment:
This derived from the new pseudocode in the SDAM spec: https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#error-handling-pseudocode